pydantic_evals/pydantic_evals/dataset.py

```diff
@@ -291,11 +307,16 @@ async def _handle_case(case: Case[InputsT, OutputT, MetadataT], report_case_name
             eval_span.set_attribute('cases', report.cases)
             # TODO(DavidM): Remove this 'averages' attribute once we compute it in the details panel
             eval_span.set_attribute('averages', report.averages())
-
+        if progress:  # pragma: no branch
+            progress_bar.close()  # pyright: ignore[reportPossiblyUnboundVariable]
```

Comment:
Can we remove the need for these pyright ignores by checking for `progress_bar` instead? You'll want to set it to `None` above `if progress:` up top, so it always has a value.